### DISCUSSION TITLE
Transform #[command] result type

### DISCUSSION THREAD

**Discussion by pronebird** [tauri-apps/tauri#9585] [Q&A]
Hi,

Would it be possible to capture the output of `#[command]` functions and do the transformation before the `serde_json::Value` is returned to browser? I need to adapt field names for camel-case and on serde side it's a bit problematic since my types come from snake_case world and there is really no way to process nested structures but re-keying the serialized representation in `serde_json::Value` .

**Comment by FabianLars** [comment]
This is currently (or never) supported so you'd either have to fix your serde setup, modify the `Value` before returning it, or maybe even serialize it to a json string and modify that string directly. (The string solution requires you to use JSON.parse on the js side though)

If that doesn't help, then a small code example would probably help us to help you :)

**Reply by pronebird** [reply]
@FabianLars Thanks for your answer. Definitely works with `serde_json::Value`, although if commands used to return `Result<ConcreteType, Error>`, now they return `Result<serde_json::Value, Error>`. That works fine except it's not as descriptive anymore.
